test/net/sourceforge/kolmafia/session/RabbitHoleManagerTest.java

```diff
@@ -0,0 +1,233 @@
+package net.sourceforge.kolmafia.session;
+
+import static internal.helpers.Networking.assertGetRequest;
+import static internal.helpers.Networking.assertPostRequest;
+import static internal.helpers.Networking.html;
+import static internal.helpers.Player.withGender;
+import static internal.helpers.Player.withHP;
+import static internal.helpers.Player.withHandlingChoice;
+import static internal.helpers.Player.withHttpClientBuilder;
+import static internal.helpers.Player.withPasswordHash;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import internal.helpers.Cleanups;
+import internal.network.FakeHttpClientBuilder;
+import java.util.List;
+import java.util.Map;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import net.sourceforge.kolmafia.request.RelayRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+public class RabbitHoleManagerTest {
+
+  @BeforeAll
+  public static void beforeAll() {
+    KoLCharacter.reset("rabbit hole");
+    RabbitHoleManager.reset();
+  }
+
+  @BeforeEach
+  public void beforeEach() {
+    Preferences.reset("rabbit hole");
+  }
+
+  @Nested
+  class Chess {
+    @AfterEach
+    public void afterEach() {
+      InventoryManager.resetInventory();
+      RelayRequest.specialCommandIsAdventure = false;
+      RelayRequest.specialCommandResponse = "";
+      RelayRequest.specialCommandStatus = "";
+      RelayRequest.redirectedCommandURL = "";
```

Comment:
Might be worth making a `RelayRequest.reset()` to group these, given they'll be wanted after each test interacting with the relay request.